tests/Aspire.Dashboard.Tests/Model/DashpageJsonFilePersistenceTests.cs

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Model;
+using Xunit;
+using static Aspire.Dashboard.Components.Pages.Metrics;
+
+namespace Aspire.Dashboard.Tests.Model;
+
+public sealed class DashpageJsonFilePersistenceTests
+{
+    [Fact]
+    public void Deserialize()
+    {
+        // Start with JSON, convert to object model, back to JSON, and compare with original.
+        var json = """
+            [
+                /* Comments are ignored */
+                {
+                    "name": ".NET",
+                    "priority": 99,
+                    "charts": [
+                        {
+                            "title": "Exception count",
+                            "instrument": "process.runtime.dotnet.exceptions.count",
+                            "required": false, // default
+                            "resource": null,  // default
+                            "kind": "Graph"    // upper case
+                        },
+                        {
+                            "title": "Assembly count",
+                            "instrument": "process.runtime.dotnet.assemblies.count",
+                            "kind": "table"    // lower case
+                        },
+                        {
+                            "title": "Thread Pool Completion",
+                            "instrument": "process.runtime.dotnet.thread_pool.completed_items.count",
+                            "required": true   // non-default
+                        }
+                    ]
+                },
+                {
+                    "name": "Envoy",
+                    "priority": 99,
+                    "charts": [
+                        {
+                            "title": "Connection count",
+                            "instrument": "envoy.connection.count",
+                            "required": true,   // non-default
+                            "resource": "envoy" // non-default
+                        }
+                    ]
+                }
+            ]
+            """;
+
+        var dashpages = DashpageJsonFilePersistence.Deserialize(json);
```

Comment:
Can the test use ` DashpageJsonFilePersistence.GetDashpagesAsync` instead since that is the public API?

Reply:
That requires writing a file or mocking the file provider which I wanted to avoid.

Reply:
Writing a json file to be loaded by the public api? It can have the same contents as in this existing test, right?

Reply:
The main thing I want to test is the transformation between string and object model. The other API tests more layers than needed here.